src/planner/ompl/CRRT.cpp

```diff
@@ -358,6 +378,18 @@ CRRT::Motion* CRRT::constrainedExtend(
       }
     }
 
+    // NOTE: (sniyaz) This check makes sure the resolution of the path on the
+    // constraint manifold and the resolution used for interpolation (*before*
+    // projection) do not differ wildly. In other words, after projection
+    // the distance between the nearest-neighbor and the projected state should
+    // lie within a scalar multiple of `mProjectionResolution`.
+    double manifoldResolution = si_->distance(xstate, cmotion->state);
+    if (manifoldResolution
+        > mFinalResolutionSlackFactor * mProjectionResolution)
+    {
+      break;
```

Comment:
Should it `break` or should it `continue` to the next random sample?

Reply:
I think the break is correct: we don't actually sample `gstate` in `constrainedExtend`, we just move towards this state that has already been sampled before `constrainedExtend` is called (see line `260` of `CRRT.cpp`).

I think this is the same reason we break on line `377`.

Reply:
Ah yes, my bad. I mistook this to be a change in `solve()`.